### ISSUE TITLE
$png-data() doesn't seem to work

### ISSUE THREAD

**Issue by alanmead** [benkasminbullock/Image-PNG#17]
I'm confused how having the image data as a scalar would work, but I want to access the individual pixel values of an image. I assume $png-data() would be the way, but I cannot get it to return anything but `undef`:

```
use strict;
use warnings;

use Image::PNG;
use Data::Dumper;

my $png = Image::PNG->new ();
my $fn = shift or die;
$png->read($fn);
printf "$fn: %d x %d\n", $png->width, $png->height;
printf "color type = %s\n", $png->color_type;
printf "bit depth = %d\n", $png->bit_depth;
#printf "interlacing method = %d\n", $png->interlace_type();
my $data = $png->data();
print Dumper( $data );
```
Produces:
```
$ ./test_png.pl yeti-males-females.png
yeti-males-females.png: 1000 x 561
color type = RGB_ALPHA
bit depth = 8
$VAR1 = undef;
```
Looking at the code, this is because nothing sets the $png->{data}; (unless the user adds data manually), so line 297 returns undef. Is this correct that $png->{data}; should not be set in Image::PNG:read()? Is line 115 a typo (it sets $png->{read} but maybe should set $png->{data})?

**Comment by alanmead** [comment]
I guess $png->rows() returns image data. Am I mistaken about what $png->data() does? (It doesn't actually "Get the PNG image data"?)

**Comment by benkasminbullock** [comment]
Thank you very much for the bug reports.

For the time being, Image::PNG::Libpng is a better bet for a module to
access the PNG library.

I would suggest switching to that module if possible.

**Comment by benkasminbullock** [comment]
Looking at the code, it doesn't seem to have ever been implemented, and it is quite a while since I did this, and I don't remember what it was going to be originally. Since the data of the image can be set using the rows method, what I think would be good would to remove it from the module and remove the documentation too, which is what I'm doing for version 0.25.

Thank you for the bug report, sorry to have wasted your or other people's time by not properly implementing the documented functions.

**Comment by alanmead** [comment]
You didn't waste my time. I'm sorry if my bug report came across that way. I appreciate your contribution to CPAN. 


-Alan 



Sent from Workspace ONE Boxer 


On Mar 20, 2021 1:25 AM, Ben Bullock ***@***.***> wrote: 

Closed #17.
